Libplanet/Blockchain/Policies/VolatileStagePolicy.cs

```diff
@@ -189,39 +189,14 @@ public bool Ignores(BlockChain<T> blockChain, TxId id)
         /// <inheritdoc/>
         public Transaction<T>? Get(BlockChain<T> blockChain, TxId id, bool filtered = true)
         {
-            Transaction<T>? transaction = null;
-
-            _lock.EnterWriteLock();
+            _lock.EnterReadLock();
```

Comment:
Write lock was used since we wanted to modify `_staged` during lookup. Are we sure this won't cause a problem?

Reply:
At least from what I checked, it would be okay because...

1. `_staged` is already `ConccurrentDictionary<K, V>` so it might be safe against multiple threads itself.
2. While `_staged` updating, I think that we wouldn't perform any other ops(i.e., `.TryUpdate()` and return null) that should be synchronized as a writer (not a reader because we're already in that scope).

Reply:
It almost seems okay also to my eyes but still...

- `_staged` being `ConcurrentDictionary<K, V>` isn't that relevant. It has to do with look-up and conditional check operations being separate from `TryRemove()` call.
- While `_staged` is updating, external context may change, i.e. `Expired(tx) || _ignored.Contains(tx.Id)`, which is affected by other calls such as (`Ignore()`).

It is not that clear that this would be a safe change that doesn't break the concurrency of `VolatileStagePolicy`'s internal states.

Reply:
> _staged being ConcurrentDictionary<K, V> isn't that relevant. It has to do with look-up and conditional check operations being separate from TryRemove() call.

Right, but nevertheless I mentioned `ConcurrentDictionary<K, V>` since plain `Dictionary<K, V>` tends to misbehave even just one operation in a multi-threaded environment. 😅 

> While _staged is updating, external context may change, i.e. Expired(tx) || _ignored.Contains(tx.Id), which is affected by other calls such as (Ignore()).

Basically, locking mechanisms are used to prevent invalid behaviors by controlling the accessibility of an intermediate state. but in this case, there are several points to consider 🤔 

1. As this logic only mutate `_staged` and the mutation against it might be atomic (due to `ConcurrentDictionary<K, V>`). it means that other viewers have no write lock or no difference.
2. At the same time, we must protect the current execution context from being affected by changes in other threads. and I guess that it's okay because we've acquired `_lock` as reader and other threads can't enter as writer.
    - Of course,  since these read-write locks are entirely based on the programmer's code, it's also possible that assumptions can be broken by improper locking. 
    - But if we assume that it won't acquire write locks properly, then all read locks in the class, not just this logic, will be meaningless.

Reply:
> At the same time, we must protect the current execution context from being affected by changes in other threads. and I guess that it's okay because we've acquired _lock as reader and other threads can't enter as writer.

But yeah.. on second thought, I may need to more consider `Get()` vs other `Get()`s... 🤔